paywall/src/data-iframe/blockchainHandler/keyStatus.js

```diff
@@ -49,14 +49,21 @@ export function linkTransactionsToKey({
   transactions,
   requiredConfirmations,
 }) {
-  const account = key.owner
+  const account = key && key.owner ? key.owner.toLowerCase() : null
+  const lock = key && key.lock ? key.lock.toLowerCase() : null
   const keyPurchaseTransactions = Object.values(transactions)
     .filter(transaction => {
       // in theory, all transactions have already been filtered this way
       // but this is a sanity check to ensure we only process
       // key purchase transactions where the key was purchased
       // for the current account holder
-      return transaction.for === account && transaction.to === key.lock
```

Comment:
we use `transaction.lock` instead of `transaction.to` because in live testing, there were transactions that lacked a `to` field, but all key purchase transactions have a `lock` field